### PR TITLE
Stack Docs

### PR DESCRIPTION
**Pull request by invokesus**
~~Merge after #3938 .~~

### CODE REVIEW

File: site/docs/transform/stack.md [hunk 1]
```diff
@@ -4,10 +4,15 @@ title: Stack
 permalink: /docs/stack.html
 ---
```
Comment (kanitw) [inline site/docs/transform/stack.md]: Missing intro paragraph

File: site/docs/transform/stack.md [hunk 2]
```diff
@@ -27,43 +32,40 @@ determines type of stacking offset if the field should be stacked.
...
+### Normalized Stacked Bar and Area Charts
 
 You can set `stack` to `"normalize"` to create normalized (or percentage) stacked bar and area charts.
 
 <div class="vl-example" data-name="stacked_bar_normalize"></div>
 
 <div class="vl-example" data-name="stacked_area_normalize"></div>
-
-## Streamgraph
+### Streamgraph
```
Comment (kanitw) [inline site/docs/transform/stack.md]: Why removing line breaks here?

File: site/docs/transform/stack.md [hunk 3]
```diff
@@ -94,8 +96,47 @@ If you want to define custom sort order, you can derive a new field using the [`
...
+     ...
+  ],
+  ...
+}
+```
+
+The `stack` transform in the `transform` array has the following properties:
+
+{% include table.html props="stack,groupby,offset,sort,as" source="StackTransform" %}
+
```
Comment (kanitw) [inline site/docs/transform/stack.md]: To make readers understand how it works, you should pick one of the simplest example from above and produce an equivalent spec with stack transfrom so users understand the idea.

You can then explain that this is useful for composing stack with other transforms such as diverging bar chart and mosaic chart.

File: site/docs/transform/stack.md [hunk 4]
```diff
@@ -94,8 +96,47 @@ If you want to define custom sort order, you can derive a new field using the [`
...
+    {"stack": ..., "groupby": ..., "offset": ..., "sort": ..., "as" ...} // Stack Transform
+     ...
+  ],
+  ...
+}
+```
+
+The `stack` transform in the `transform` array has the following properties:
+
+{% include table.html props="stack,groupby,offset,sort,as" source="StackTransform" %}
```
Comment (kanitw) [inline site/docs/transform/stack.md]: Somehow the table is still empty

image

File: site/docs/transform/stack.md [hunk 5]
```diff
@@ -94,8 +96,47 @@ If you want to define custom sort order, you can derive a new field using the [`
...
+{:#transform}
+## Stack Transform
+
+{: .suppress-error}
+```json
+// A View Specification
+{
+  ...
+  "transform": [
+    {"stack": ..., "groupby": ..., "offset": ..., "sort": ..., "as" ...} // Stack Transform
```
Comment (kanitw) [inline site/docs/transform/stack.md]: Please expand this object to be on multiple lines (one property per line)

File: site/docs/transform/stack.md [hunk 6]
```diff
@@ -94,8 +96,47 @@ If you want to define custom sort order, you can derive a new field using the [`
...
+The `stack` transform in the `transform` array has the following properties:
+
+{% include table.html props="stack,groupby,offset,sort,as" source="StackTransform" %}
+
+### Diverging Bar Chart
+
+Here we initially `stack` by `"question"` and then use the `window` transform to offset each stack.
+<div class="vl-example" data-name="bar_diverging_stack_transform"></div>
+
+### Basic Mosaic Chart
```
Comment (kanitw) [inline site/docs/transform/stack.md]: 1) I don't know if we need to show all of these variant of mosaic chart examples.

File: site/docs/transform/stack.md [hunk 7]
```diff
@@ -94,8 +96,47 @@ If you want to define custom sort order, you can derive a new field using the [`
...
+```
+
+The `stack` transform in the `transform` array has the following properties:
+
+{% include table.html props="stack,groupby,offset,sort,as" source="StackTransform" %}
+
+### Diverging Bar Chart
+
+Here we initially `stack` by `"question"` and then use the `window` transform to offset each stack.
+<div class="vl-example" data-name="bar_diverging_stack_transform"></div>
```
Comment (kanitw) [inline site/docs/transform/stack.md]: Is `bar_diverging_stack_transform` the same example as in the gallery?

File: site/docs/transform/stack.md (hunk 6 above)
Comment (kanitw) [reply site/docs/transform/stack.md]: For 2., I'm asking @domoritz  to add `json-stringify-pretty-compact` to  bit.ly/js2json -- so we can use it.

Comment (invokesus) [reply site/docs/transform/stack.md]: I kept only the simple mosaic example. Will make a separate PR for adding `rect_mosaic_labelled_with_offset`to the gallery and then add a link to it.

File: site/docs/transform/stack.md (hunk 7 above)
Comment (invokesus) [reply site/docs/transform/stack.md]: I'll make a separate PR for replacing the gallery example.

Comment (kanitw) [reply site/docs/transform/stack.md]: I think it should be done within the scope of this PR.

File: site/docs/transform/stack.md (hunk 6 above)
Comment (kanitw) [reply site/docs/transform/stack.md]: Let's just do it here.

File: _data/examples.json [hunk 8]
```diff
@@ -46,7 +46,7 @@
         "title": "Layered Bar Chart"
       },
       {
-        "name": "bar_diverging_stacked",
```
Comment (kanitw) [inline _data/examples.json]: Please remove the actual example too as it's not needed anymore

**Comment by kanitw** [comment]
Please also just do this here:
I kept only the simple mosaic example. Will make a separate PR for adding rect_mosaic_labelled_with_offsetto the gallery and then add a link to it.

**Pull request merged**